src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -22,6 +24,26 @@
 from pants.rules.core.strip_source_root import SourceRootStrippedSources
 
 
+def calculate_timeout_seconds(
+  *,
+  timeouts: bool,
+  test_target_timeout_seconds: Optional[int],
```

Comment:
Consider dropping `test_`. It's an implementation detail that this is a test target. Really, the main idea is that the number comes from the target, rather than from a global option, and I think `target_timeout` would express that.